osu.Game.Rulesets.Taiko/Mods/TaikoModAlternate.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.Taiko.Objects;
+
+namespace osu.Game.Rulesets.Taiko.Mods
+{
+    public class TaikoModAlternate : ModAlternate<TaikoHitObject, TaikoAction>
+    {
+        [SettingSource("Playstyle", "Preferred Alternate Playstyle")]
+        public Bindable<Playstyle> Style { get; } = new Bindable<Playstyle>();
+
+        [SettingSource("Allow any key after drum roll")]
+        public BindableBool ResetAfterDrumRoll { get; } = new BindableBool();
+
+        private TaikoAction? prevPressedAction;
+        private TaikoHitObject prevHitObject;
+        private double prevActionTime;
+        private bool strongHitSucceeded;
+        private const double strong_hit_window = 30;
+
+        protected override void ResetActionStates()
+        {
+            prevPressedAction = null;
+        }
+
+        protected override bool OnPressed(TaikoAction action)
+        {
+            if (IsBreakTime.Value)
+                return false;
+
+            var hitObject = HitObjects.FirstOrDefault(h =>
+            {
+                var time = Interceptor.Time.Current;
+                var window = (h is DrumRoll) ? ((h.NestedHitObjects.FirstOrDefault() as DrumRollTick)?.HitWindow ?? 0) : h.HitWindows.WindowFor(HitResult.Miss);
+                var endTime = (h as IHasDuration)?.EndTime ?? h.StartTime;
+                return (time > h.StartTime - window) && (time < endTime + window);
```

Comment:
i don't even know where to start here. extract to a method and explain what is going on. and don't use var for primitive types.